resources/user-agents/browsers/firefox-on-firefoxos/firefox-os-1-1.json

```diff
@@ -28,36 +28,30 @@
       },
       "children": [
         {
-          "match": "Mozilla/5.0 (Mobile; ALCATELOneTouch4012X*; rv:#MAJORVER#.#MINORVER#*)*Gecko/#MAJORVER#.#MINORVER#*Firefox/#MAJORVER#.#MINORVER#*",
-          "device": "Alcatel OT-4012X"
-        },
-        {
-          "match": "Mozilla/5.0 (Mobile; ALCATEL ONE TOUCH 4012A; rv:#MAJORVER#.#MINORVER#*)*Gecko/#MAJORVER#.#MINORVER#*Firefox/#MAJORVER#.#MINORVER#*",
-          "device": "Alcatel OT-4012A"
-        },
-        {
-          "match": "Mozilla/5.0 (Mobile; LG-D300; rv:#MAJORVER#.#MINORVER#*)*Gecko/#MAJORVER#.#MINORVER#*Firefox/#MAJORVER#.#MINORVER#*",
-          "device": "LG D300"
-        },
-        {
-          "match": "Mozilla/5.0 (Mobile; ZTEOPEN; rv:#MAJORVER#.#MINORVER#*)*Gecko/#MAJORVER#.#MINORVER#*Firefox/#MAJORVER#.#MINORVER#*",
-          "device": "ZTE Open"
+          "match": "Mozilla/5.0 (Mobile; #DEVICE#; rv:#MAJORVER#.#MINORVER#*)*Gecko/#MAJORVER#.#MINORVER#*Firefox/#MAJORVER#.#MINORVER#*",
+          "devices": {
+            "ALCATELOneTouch4012X*": "Alcatel OT-4012X",
+            "ALCATEL ONE TOUCH 4012A": "Alcatel OT-4012A",
+            "ALCATELOneTouch4012A": "Alcatel OT-4012A",
+            "LG-D300": "LG D300",
+            "ZTEOPEN": "ZTE Open"
+          }
         },
         {
-          "match": "Mozilla/5.0 (Mobile; rv:#MAJORVER#.#MINORVER#*)*Gecko/#MAJORVER#.#MINORVER#*Firefox/#MAJORVER#.#MINORVER#*"
+          "match": "Mozilla/5.0 (Mobile; *rv:#MAJORVER#.#MINORVER#*)*Gecko/#MAJORVER#.#MINORVER#*Firefox/#MAJORVER#.#MINORVER#*"
```

Comment:
I added the asterisk here to allow for these patterns to catch other devices that aren't in browscap yet, so at least they're being caught as Firefox on Firefox OS instead of Default Browser.

There aren't a lot of devices that are compatible with Firefox OS, but without seeing the agents in the wild, I don't know what their device identifiers are.

https://en.wikipedia.org/wiki/Comparison_of_Firefox_OS_devices